### DISCUSSION TITLE
Do long polling requests from clients ever expire or stop attempting to connect

### DISCUSSION THREAD

**Discussion by RyanAP1** [socketio/socket.io#4847] [General] [answered]
Long story short, I have a web app where I was using socket.io at first, but transitioned away from it as I found a stateless way to handle my workloads. I used socket.io it for many months before transitioning away and even after about two days of having no socket.io references in my backend or front end code, I still see hundreds of long-polling requests 404ing against my application in production

My best guess is that clients who still had my page open are still attempting to connect, but I would have imagined that after a couple minutes all those requests would expire or stop attempting to connect.

Is there a built in expiration or retry limit on clients attempting to connect to the socket.io server(which is no longer there)?

**Comment by darrachequesne** [comment, marked as answer]
Hi!

The `reconnectionAttempts` option defaults to `Infinity`, so I think unfortunately those clients will try to reconnect forever.

Reference: https://socket.io/docs/v4/client-options/#reconnectionattempts

**Reply by RyanAP1** [reply]
Thanks for the answer. Looks like I'll either have to re-add some components to disconnect clients myself or wait for them to taper off over time from restarts or cache clears.
